hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/io/storage/row/parquet/ParquetSchemaConverter.java

```diff
@@ -600,23 +600,29 @@ private static Type convertToParquetType(
             .named(name);
       case TIMESTAMP_WITHOUT_TIME_ZONE:
         TimestampType timestampType = (TimestampType) type;
-        if (timestampType.getPrecision() == 3) {
+        if (timestampType.getPrecision() == 3 || timestampType.getPrecision() == 6) {
+          TimeUnit timeunit = timestampType.getPrecision() == 3 ? TimeUnit.MILLIS : TimeUnit.MICROS;
           return Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, repetition)
-              .as(LogicalTypeAnnotation.timestampType(true, TimeUnit.MILLIS))
+              .as(LogicalTypeAnnotation.timestampType(true, timeunit))
               .named(name);
         } else {
-          return Types.primitive(PrimitiveType.PrimitiveTypeName.INT96, repetition)
-              .named(name);
+          throw new IllegalArgumentException(
+              "Parquet does not support TIMESTAMP type with precision: "
+                  + timestampType.getPrecision()
+                  + ", it only support precisions <= 6.");
         }
       case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
         LocalZonedTimestampType localZonedTimestampType = (LocalZonedTimestampType) type;
-        if (localZonedTimestampType.getPrecision() == 3) {
+        if (localZonedTimestampType.getPrecision() == 3 || localZonedTimestampType.getPrecision() == 6) {
+          TimeUnit timeunit = localZonedTimestampType.getPrecision() == 3 ? TimeUnit.MILLIS : TimeUnit.MICROS;
           return Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, repetition)
-              .as(LogicalTypeAnnotation.timestampType(false, TimeUnit.MILLIS))
+              .as(LogicalTypeAnnotation.timestampType(false, timeunit))
               .named(name);
         } else {
-          return Types.primitive(PrimitiveType.PrimitiveTypeName.INT96, repetition)
-              .named(name);
+          throw new IllegalArgumentException(
+              "Parquet does not support TIMESTAMP_LTZ type with precision: "
+                  + localZonedTimestampType.getPrecision()
```

Comment:
Revert the change to not throw exception.

Reply:
Done